numpyro/primitives.py

```diff
@@ -33,6 +33,10 @@ def apply_stack(msg):
         else:
             msg['value'] = msg['fn'](*msg['args'], **msg['kwargs'])
 
+        if msg['type'] == 'param' and callable(msg['value']):
+            rng_key = prng_key()
```

Comment:
What is `prng_key`?

Reply:
it is https://github.com/pyro-ppl/numpyro/pull/804. :D

Reply:
Haha..I completely blanked out on the name and didn't see the definition here.

Reply:
How about adding this after the `if` clause:
```python
if msg['type'] == 'sample':
  ...
elif msg['type'] == 'param':
  ...
else:
  ...
```

Actually, I'm thinking that if you modify the `fn` in the `init_msg` for `param` to store the initialization function and pass the rng_key to args, we may not even need to change anything here.

Reply:
Thanks, I just moved the logic to param primitive. It seems cleaner.